python/pyarrow/pandas-shim.pxi

```diff
@@ -183,6 +186,28 @@ cdef class _PandasAPIShim(object):
         else:
             return False
 
+    cpdef is_index(self, obj):
+        if self._have_pandas_internal():
```

Comment:
I'm curious, why is this necessary? Perhaps add a comment?

Reply:
All the other `is_..` methods above use the same pattern. 
The reason it is needed is because this method can in principle be called in a situation that pandas is not installed. So therefore we first check that with `self._have_pandas_internal` (the "internal" version is a cdef), and otherwise always return False.

Reply:
Thanks for the explanation.